py/server/deephaven/table_factory.py

```diff
@@ -272,6 +285,70 @@ def delete(self, table: Table) -> None:
         except Exception as e:
             raise DHError(e, "delete data in the InputTable failed.") from e
 
+    def add_async(self, table: Table, on_success: Callable[[], None] = None,
```

Comment:
support should also be added to the client

Reply:
I am not sure that we need to.
1. async add/delete is what the gRPC API uses already on the server. 
2. the Python client talks to the server only synchronously. It'd be a paradigm shift if we were to support Python asyncIO, not to say it is not doable.

Reply:
My concern on this is that users frequently want to write code once and use it on both the client and server.  If that might happen with `InputTable`, we should have the method to make the API consistent, even if the method is just calling `add`.  If there are no reasonable cases where that might happen, then maybe it isn't a concern.  This is what I'm worried about.

Reply:
I don't have a problem with introducing async support, but I do think Jianfeng is right that it may significantly expand scope and require some research on his end. I'd rather we do that in a separate PR, since this one is self-contained and makes things better.

Reply:
I'm ok with a separate PR.